File: data/EXP01/python/2564/2564.output_2.py
# @lc app=leetcode id=2564 lang=python3
from typing import List

class Solution:
    def substringXorQueries(self, s: str, queries: List[List[int]]) -> List[List[int]]:
        xorArr = [0]
        for i in range(len(s)):
            xorArr.append(xorArr[-1] ^ int(s[i]))

        xorMap = {}
        for i in range(len(xorArr)):
            if xorArr[i] not in xorMap:
                xorMap[xorArr[i]] = i

        result = []
        for query in queries:
            first, second = query
            if second not in xorMap:
                result.append([-1, -1])
            else:
                start = xorMap[second]
                end = -1

                for i in range(start, len(xorArr)):
                    if xorArr[i] ^ xorArr[start - 1] == first:
                        end = i
                        break
                
                result.append([start - 1, end - 1])

        return result

File: data/EXP01/golang_no_system_var/2564/2564.output_2.go
func substringXorQueries(s string, queries [][]int) [][]int {
    n := len(s)
    ans := make([][]int, len(queries))
    
    prefixXor := make([]int, n+1)
    for i := 1; i <= n; i++ {
        prefixXor[i] = prefixXor[i-1] ^ int(s[i-1]-'0')
    }
    
    for i, query := range queries {
        left := query[0]
        right := query[1]
        
        for j := left; j <= right; j++ {
            for k := j+1; k <= right; k++ {
                xor := prefixXor[k] ^ prefixXor[j]
                if xor == query[0] {
                    ans[i] = []int{j, k-1}
                    break
                }
            }
            
            if ans[i] != nil {
                break
            }
        }
        
        if ans[i] == nil {
            ans[i] = []int{-1, -1}
        }
    }
    
    return ans
}
